### PR TITLE
db/doclets-component-options

### PR DESCRIPTION
**Pull request by sebastianbochan**

### CODE REVIEW

**Review by bre1470 — APPROVED** [review]
Looks good! 👍

File: ts/Dashboards/Component/Component.ts [hunk 1]
```diff
@@ -981,7 +1028,14 @@ namespace Component {
     * */
 
     /**
+     * Adds component to the registry.
      *
+     * @param {T} componentClass
```
Comment (karolkolodziej) [inline ts/Dashboards/Component/Component.ts:1033]: With the typedoc there is no need to specify the type as you did `{T}`

File: ts/Dashboards/Component/Component.ts [hunk 2]
```diff
@@ -981,7 +1028,14 @@ namespace Component {
...
+     * Component class.
+     *
+     * @return {boolean}
```
Comment (karolkolodziej) [inline ts/Dashboards/Component/Component.ts:1036]: Same here with type

File: ts/Dashboards/Component/Component.ts [hunk 3]
```diff
@@ -1001,14 +1055,20 @@ namespace Component {
     }
 
     /**
+     * Extract all components names from the registry.
      *
+     * @return {Array<string>}
```
Comment (karolkolodziej) [inline ts/Dashboards/Component/Component.ts:1060]: Please check each file for issue with types

File: ts/Dashboards/Component/Component.ts [hunk 4]
```diff
@@ -1001,14 +1055,20 @@ namespace Component {
...
+     * Returns the array of components names.
      */
     export function getAllComponentNames(): Array<string> {
```
Comment (karolkolodziej) [inline ts/Dashboards/Component/Component.ts:1063]: This `getAllComponentNames` isn't used anywhere. Do we need it?

File: ts/Dashboards/Component/Component.ts [hunk 5]
```diff
@@ -1001,14 +1055,20 @@ namespace Component {
...
         return Object.keys(Component.registry);
     }
 
     /**
+     * Extract all components from the registry.
      *
+     * @return {Record<string, Class<Component>>}
+     * Returns the array of components.
      */
     export function getAllComponents(): Record<string, Class<Component>> {
```
Comment (karolkolodziej) [inline ts/Dashboards/Component/Component.ts:1073]: It isn't used anywhere. Do we need it?

File: ts/Dashboards/Component/HTMLComponent.ts [hunk 6]
```diff
@@ -273,20 +273,44 @@ class HTMLComponent extends Component<HTMLComponent.HTMLComponentEvents> {
...
+    *
+    * */
+
     export type ComponentType = HTMLComponent;
     export interface HTMLComponentOptions extends Component.ComponentOptions, EditableOptions {
+        /**
+         * Array of HTML elements, declared as string or node.
+         *
+         * ```
+         * Example:
```
Comment (karolkolodziej) [inline ts/Dashboards/Component/HTMLComponent.ts:294]: Move `example` one line up

File: ts/Extensions/DashboardPlugins/HighchartsComponent.ts [hunk 7]
```diff
@@ -602,25 +608,46 @@ namespace HighchartsComponent {
     }>;
 
     export interface ComponentOptions extends Component.ComponentOptions, EditableOptions {
+        /**
+         * Declares update of the store, when edit chart.
```
Comment (karolkolodziej) [inline ts/Extensions/DashboardPlugins/HighchartsComponent.ts:612]: Please check the description where the default value is set, update this and remove the one on the top.

File: ts/Extensions/DashboardPlugins/HighchartsComponent.ts [hunk 8]
```diff
@@ -602,25 +608,46 @@ namespace HighchartsComponent {
...
+         */
         allowStoreUpdate?: boolean,
+        /**
+         * The string that declares contructor that is called for creating
```
Comment (karolkolodziej) [inline ts/Extensions/DashboardPlugins/HighchartsComponent.ts:616]: typo `contructor`

File: ts/Extensions/DashboardPlugins/DataGridComponent.ts [hunk 9]
```diff
@@ -356,23 +363,42 @@ namespace DataGridComponent {
...
-
+    /** @internal */
     export interface EditableOptions extends Component.EditableOptions {
+        /**
+         * The set of options like `dataGridClassName` and `dataGridID`
+         */
         dataGridOptions?: DataGridOptions;
+        /**
+         * The name of class that is applied to the chart's container
+         */
```
Comment (karolkolodziej) [inline ts/Extensions/DashboardPlugins/DataGridComponent.ts:384]: Should be renamed to `dataGridClassName` and `dataGridID`

**Review by karolkolodziej — CHANGES_REQUESTED** [review]

File: ts/Dashboards/Component/Component.ts [hunk 10]
```diff
@@ -923,17 +955,32 @@ namespace Component {
...
+         * Sets an ID for the component's `div`
+         */
         id?: string;
+        /**
+         * Additional CSS styles to apply inline to the component `div`
+         */
         style?: CSSObject;
+        /**
+         * The component's title, which will render at the top
+         */
```
[the remark this answers is not in the document]
Comment (sebastianbochan) [reply ts/Dashboards/Component/Component.ts:974]: @goransle ?

File: ts/Dashboards/Component/Component.ts [hunk 11]
```diff
@@ -923,17 +955,32 @@ namespace Component {
...
         title: TextOptionsType;
+        /**
+         * The component's caption, which will render at the bottom
+         */
```
[the remark this answers is not in the document]
Comment (sebastianbochan) [reply ts/Dashboards/Component/Component.ts:978]: @goransle ?

File: ts/Dashboards/Component/Component.ts (hunk 5 above)
Comment (sebastianbochan) [reply ts/Dashboards/Component/Component.ts:1073]: @goransle ?

File: ts/Dashboards/Component/Component.ts (hunk 4 above)
Comment (sebastianbochan) [reply ts/Dashboards/Component/Component.ts:1063]: @goransle ?

File: ts/Dashboards/Component/Component.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (goransle) [reply ts/Dashboards/Component/Component.ts:974]: Should be no issue with making these optional

File: ts/Dashboards/Component/Component.ts (hunk 5 above)
Comment (goransle) [reply ts/Dashboards/Component/Component.ts:1073]: No opinion. It's meant to be used in with es6 imports and in a "dashboardless" setup

File: ts/Dashboards/Component/Component.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (sebastianbochan) [reply ts/Dashboards/Component/Component.ts:977]: I converted it to optional, all worked properly :)

File: ts/Dashboards/Component/Component.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (sebastianbochan) [reply ts/Dashboards/Component/Component.ts:973]: I converted it to optional, all worked properly :)

File: ts/Dashboards/Component/Component.ts [hunk 12]
```diff
@@ -899,20 +899,52 @@ namespace Component {
...
         editableOptions: Array<string>;
+        /** @internal */
         editableOptionsBindings: EditableOptions.OptionsBindings;
+        /** @internal */
         presentationModifier?: DataModifier;
+        /**
+         * Defines which elements should be synced.
+         *
+         * ```
+         * Example:
```
[the remark this answers is not in the document]
Comment (karolkolodziej) [reply ts/Dashboards/Component/Component.ts:935]: not resolved

File: ts/Dashboards/Component/HTMLComponent.ts (hunk 6 above)
Comment (karolkolodziej) [reply ts/Dashboards/Component/HTMLComponent.ts:293]: same

File: ts/Extensions/DashboardPlugins/HighchartsComponent.ts [hunk 13]
```diff
@@ -333,12 +328,12 @@ class HighchartsComponent extends Component<HighchartsComponent.ChartComponentEv
         table.setCell(columnName, rowNumber, valueToSet);
     }
     /**
-     * Handles updating via options
+     * Handles updating via options.
```
Comment (karolkolodziej) [inline ts/Extensions/DashboardPlugins/HighchartsComponent.ts:331]: Please remove the params types from each file

**Review by karolkolodziej — CHANGES_REQUESTED** [review]

File: ts/Extensions/DashboardPlugins/HighchartsComponent.ts [hunk 14]
```diff
@@ -333,12 +328,12 @@ class HighchartsComponent extends Component<HighchartsComponent.ChartComponentEv
...
      * @param {Partial<Component.ComponentOptions>} options
```
Comment (karolkolodziej) [inline ts/Extensions/DashboardPlugins/HighchartsComponent.ts:332]: You still have the type (`{Partial<Component.ComponentOptions>}`) in brackets here- please remove it.
Also check other files/part of the code for the same.

**Review by karolkolodziej — CHANGES_REQUESTED** [review]

File: ts/Extensions/DashboardPlugins/HighchartsComponent.ts (hunk 14 above)
Comment (sebastianbochan) [reply ts/Extensions/DashboardPlugins/HighchartsComponent.ts:332]: This is a part of the class. It's fixed in a separate ticket/PR here https://github.com/highcharts/highcharts/pull/18632 that covers classes.

File: ts/Dashboards/Components/Component.ts [hunk 15]
```diff
@@ -899,20 +899,51 @@ namespace Component {
     export type SyncOptions = Record<string, boolean | Partial<Sync.OptionsEntry>>;
 
     export interface ComponentOptions extends EditableOptions {
+        /**
+         * Instance of cell, where component is attached.
```
Comment (TorsteinHonsi) [inline ts/Dashboards/Components/Component.ts:903]: "Instance of the cell where the component is attached"

File: ts/Dashboards/Components/Component.ts [hunk 16]
```diff
@@ -899,20 +899,51 @@ namespace Component {
...
+         */
         parentCell?: Cell;
+        /**
+         * The HTML element or id of HTML element that is used for appending
+         * a component.
+         */
         parentElement: HTMLElement | string;
+        /**
+         * The name of class that is applied to the component's container.
```
Comment (TorsteinHonsi) [inline ts/Dashboards/Components/Component.ts:912]: _the_ class

File: ts/Dashboards/Components/Component.ts [hunk 17]
```diff
@@ -899,20 +899,51 @@ namespace Component {
...
         className?: string;
+        /**
+         * The type of component like: `HTML`, `KPI`, `Highcharts`, `DataGrid`.
+         */
         type: string;
         // allow overwriting gui elements
+        /** @internal */
         navigationBindings?: NavigationBindingsOptionsObject[];
+        /**
+         * Events attached to the component : `mount`, `unmount`.
```
Comment (TorsteinHonsi) [inline ts/Dashboards/Components/Component.ts:923]: No space before the colon

File: ts/Extensions/DashboardPlugins/DataGridComponent.ts [hunk 18]
```diff
@@ -356,23 +363,34 @@ namespace DataGridComponent {
     export interface DataGridOptions
         extends Component.ComponentOptions,
         EditableOptions {
+        /**
+         * The name of class that is applied to the data grid container.
```
Comment (TorsteinHonsi) [inline ts/Extensions/DashboardPlugins/DataGridComponent.ts:367]: _a_ class

File: ts/Extensions/DashboardPlugins/DataGridComponent.ts [hunk 19]
```diff
@@ -356,23 +363,34 @@ namespace DataGridComponent {
...
+         */
         dataGridClassName?: string;
+        /**
+         * The name of id that is applied to the data grid container.
```
Comment (TorsteinHonsi) [inline ts/Extensions/DashboardPlugins/DataGridComponent.ts:371]: _the_ id

File: ts/Extensions/DashboardPlugins/HighchartsComponent.ts [hunk 20]
```diff
@@ -602,25 +603,47 @@ namespace HighchartsComponent {
...
     }
-
+    /** @internal */
     export interface EditableOptions extends Component.EditableOptions {
+        /**
+         * A full set of chart options used by the chart.
+         */
         chartOptions?: Options;
+        /**
+         * The name of class that is applied to the chart's container.
```
Comment (TorsteinHonsi) [inline ts/Extensions/DashboardPlugins/HighchartsComponent.ts:627]: _a_ class

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
Some inline comments for language. 

Side question - none of the options have `@sample`'s. I assume this is another task?

**Comment by sebastianbochan** [comment]
> Some inline comments for language.
> 
> Side question - none of the options have `@sample`'s. I assume this is another task?

We will add samples / demos later.

**Review by pawelfus — CHANGES_REQUESTED** [review]
Do we need all of this in API? For example `innerResizeTimeouts`
Zrzut ekranu 2023-03-15 o 09 56 33

**Comment by sebastianbochan** [comment]
All doclets are fixed here: https://github.com/highcharts/highcharts/pull/18722

**Pull request closed without merging**